doc/news/changes/minor/20170726AlbertoSartori

```diff
@@ -0,0 +1,3 @@
+Fixed: FunctionParser uses shared_ptr instead of unique_ptr in order to fix compilation issues with gcc-5.4.
```

Comment:
This change log entry seems to be outdated. In the end, we realized that TBB was guilty and not gcc-5.4. Additionally, we only switch conditionally from a `unique_ptr` to a `shared_ptr`